src/pages/Dashboard/GatewaysInNetworkPanel.tsx

```diff
@@ -0,0 +1,155 @@
+import Placeholder from '@src/components/Placeholder';
+import Streak from '@src/components/Streak';
+import useGatewaysPerEpoch from '@src/hooks/useGatewaysPerEpoch';
+import { useEffect, useState } from 'react';
+import {
+  Area,
+  AreaChart,
+  CartesianGrid,
+  ResponsiveContainer,
+  Tooltip,
+  TooltipProps,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import {
+  NameType,
+  ValueType,
+} from 'recharts/types/component/DefaultTooltipContent';
+
+const CustomTooltip = ({
+  active,
+  payload,
+  label,
+}: TooltipProps<ValueType, NameType>) => {
+  if (active && payload && payload.length) {
+    return (
+      <div className="rounded border border-grey-500 bg-containerL0 px-4 py-2 text-mid">
+        <p>{`Epoch ${label}`}</p>
+        <p>{`Number of Gateways: ${Number(payload[0].value)}`}</p>
+      </div>
+    );
+  }
+
+  return null;
+};
+
+const GatewaysInNetworkPanel = () => {
+  const { data: gatewaysPerEpoch } = useGatewaysPerEpoch();
+
+  const [activeIndex, setActiveIndex] = useState<number>();
+
+  const [percentageChange, setPercentageChange] = useState<number>();
+
+  useEffect(() => {
+    if (gatewaysPerEpoch) {
+      setActiveIndex(gatewaysPerEpoch.length - 1);
+    }
+  }, [gatewaysPerEpoch]);
+
+  useEffect(() => {
+    if (!activeIndex || !gatewaysPerEpoch) {
+      setPercentageChange(undefined);
+    } else {
+      const currentGateways =
+        gatewaysPerEpoch[activeIndex].totalEligibleGateways;
+      const previousGateways =
+        gatewaysPerEpoch[activeIndex - 1].totalEligibleGateways;
+
+      const percentageChange =
+        ((currentGateways - previousGateways) / previousGateways) * 100;
+      setPercentageChange(percentageChange || undefined);
+    }
+  }, [activeIndex, gatewaysPerEpoch]);
```

Comment:
should be able to combine these hook calls

Reply:
I think these need to be separated. The first acts kind of like an initialization step when the gatewaysPerEpoch first load to get the activeIndex to be set to the last epoch. The second reacts to changes in activeIndex to check if a percentageChange bubble needs to be shown or not.